### PR TITLE
BoxEdit/BoxView/BoxCreate: Add Colors to the Tag Options

### PR DESCRIPTION
**Pull request by vahidbazzaz**
The PR adds colors to tag options for tags that already have colors.

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[2.0 BoxEdit: give tags color](https://trello.com/c/2McAE5b1)

**Review by pylipp — CHANGES_REQUESTED** [review]
Code review and functional review done, two remarks in comments :)

File: react/src/components/Form/SelectField.tsx [hunk 1]
```diff
@@ -62,9 +63,10 @@ function SelectField({
                 borderRadius: "0",
                 borderColor: "black",
               }),
-              multiValue: (provided) => ({
+              multiValue: (provided, state) => ({
                 ...provided,
                 border: "1px",
+                background: state.data?.color || "gray.100",
```
[the remark this answers is not in the document]
Comment (vahidbazzaz) [reply react/src/components/Form/SelectField.tsx:69]: @pylipp That makes total sense, as the user just selects the background color, then we adjust the font color using contrast ratio math. I'll push the changes.

File: react/src/components/Form/SelectField.tsx [hunk 2]
```diff
@@ -62,9 +63,10 @@ function SelectField({
                 borderRadius: "0",
                 borderColor: "black",
               }),
-              multiValue: (provided) => ({
+              multiValue: (provided, state) => ({
                 ...provided,
                 border: "1px",
```
[the remark this answers is not in the document]
Comment (vahidbazzaz) [reply react/src/components/Form/SelectField.tsx:68]: @pylipp If the background color is so bright, a border with higher contrast would make sense. I will push the changes, and you can try them out.

File: react/src/utils/helpers.ts [hunk 3]
```diff
@@ -61,3 +61,12 @@ export const generateDropappUrl = (
 export const redirectToExternalUrl = (url) => {
   window.location.replace(url);
 };
+
+export const getContrastYIQ = (hex) => {
```
Comment (pylipp) [inline react/src/utils/helpers.ts:65]: Can you add a comment, e.g.
Return true if given RGB hex color is bright, false if dark.
This is achieved by converting the RGB value into YIQ and then checking the Y component (luma). Cf. https://stackoverflow.com/a/946734/3865876

Comment (pylipp) [reply react/src/utils/helpers.ts:65]: Maybe the function name `colorIsBright` would suit better then?

File: react/src/views/BoxCreate/components/BoxCreate.tsx [hunk 4]
```diff
@@ -128,7 +128,7 @@ function BoxCreate({
   const tagsForDropdownGroups: Array<IDropdownOption> | undefined = allTags?.map((tag) => ({
     label: tag.label,
     value: tag.value,
-    // colorScheme: "red",
+    color: tag.color,
```
Comment (pylipp) [inline react/src/views/BoxCreate/components/BoxCreate.tsx:131]: Good catch :+1:

**Review by pylipp — APPROVED** [review]
Works and brings UX enhancements, nice! :tada:

**Pull request merged**